src/coreclr/jit/rangecheck.cpp

```diff
@@ -1049,6 +1049,25 @@ Range RangeCheck::ComputeRangeForBinOp(BasicBlock* block, GenTreeOp* binop, bool
                 icon = binop->OperIs(GT_RSH) ? (icon1 >> icon2) : (icon1 << icon2);
             }
         }
+        else if (binop->OperIs(GT_RSZ))
+        {
+            // (x u>> cns) -> [0..(x's max value >> cns)]
+            int shiftBy = static_cast<int>(op2->AsIntCon()->IconValue());
+            if (shiftBy < 0)
+            {
+                return Range(Limit::keUnknown);
+            }
+
+            int op1Width = (int)(genTypeSize(op1) * BITS_PER_BYTE);
+            if (shiftBy >= op1Width)
+            {
+                return Range(Limit(Limit::keConstant, 0));
+            }
+
+            // Calculate max possible value of op1, e.g. UINT_MAX for TYP_INT/TYP_UINT
```

Comment:
Was wondering if we could use `IntegralRange` here but that seems a bit clunky.

Reply:
Initially, I wanted to call `RangeCheck::ComputeRange` on op1 to get even more precise range, but it had no diffs